src/BuiltInTools/dotnet-watch/Aspire/AspireServiceFactory.cs

```diff
@@ -0,0 +1,137 @@
+
+            var projectOptions = GetProjectOptions(projectLaunchInfo);
+
+            var processTerminationSource = new CancellationTokenSource();
+
+            var runningProject = await projectLauncher.TryLaunchProcessAsync(projectOptions, processTerminationSource, build: false, cancellationToken);
+            if (runningProject == null)
+            {
+                // detailed error already reported:
+                throw new ApplicationException($"Failed to launch project '{projectLaunchInfo.ProjectPath}'.");
+            }
+
+            string sessionId;
+            lock (_guard)
+            {
+                sessionId = _sessionIdDispenser++.ToString(CultureInfo.InvariantCulture);
+                _sessions.Add(sessionId, runningProject);
+            }
+
+            Reporter.Verbose($"Session started: {sessionId}");
+            return sessionId;
+        }
+
+        /// <summary>
+        /// Implements https://github.com/dotnet/aspire/blob/445d2fc8a6a0b7ce3d8cc42def4d37b02709043b/docs/specs/IDE-execution.md#stop-session-request.
+        /// </summary>
+        public async ValueTask<bool> StopSessionAsync(string dcpId, string sessionId, CancellationToken cancellationToken)
+        {
+            Reporter.Verbose($"Stop Session {sessionId}", MessageEmoji);
+
+            RunningProject? runningProject;
+            lock (_guard)
+            {
+                if (!_sessions.TryGetValue(sessionId, out runningProject))
+                {
+                    return false;
+                }
+            }
+
+            _ = await projectLauncher.TerminateProcessesAsync([runningProject.ProjectNode.ProjectInstance.FullPath], cancellationToken);
+            return true;
+        }
+
+        private ProjectOptions GetProjectOptions(ProjectLaunchRequest projectLaunchInfo)
+        {
+            var arguments = new List<string>
+            {
+                "--project",
+                projectLaunchInfo.ProjectPath,
+                // TODO: Need to suppress launch profile for now, otherwise it would override the port set via env variable.
+                "--no-launch-profile",
+            };
+
+            //if (projectLaunchInfo.DisableLaunchProfile)
+            //{
+            //    arguments.Add("--no-launch-profile");
+            //}
+            //else if (!string.IsNullOrEmpty(projectLaunchInfo.LaunchProfile))
+            //{
+            //    arguments.Add("--launch-profile");
+            //    arguments.Add(projectLaunchInfo.LaunchProfile);
+            //}
+
+            if (projectLaunchInfo.Arguments != null)
+            {
+                arguments.AddRange(projectLaunchInfo.Arguments);
+            }
+
+            return new()
+            {
+                IsRootProject = false,
+                ProjectPath = projectLaunchInfo.ProjectPath,
+                WorkingDirectory = projectLauncher.EnvironmentOptions.WorkingDirectory, // TODO: Should DCP protocol specify?
```

Comment:
How would we get an answer on these 'TODO' comments?

Reply:
@davidfowl